src/lib/mina_base/account_id.mli

```diff
@@ -46,6 +46,8 @@ module Digest : sig
 
     val of_field : Pickles.Impls.Step.Field.t -> t
 
+    val to_field_unsafe : t -> Pickles.Impls.Step.Field.t
```

Comment:
This feels super hacky, does this work? I need a way to convert a `Mina_base.Token_id.Checked.t`  to a `Field.t`. This type checks but I'm pretty sure the logic is wrong. If there is a better way, can someone advise how to do so?

cc @mitschabaude @bkase